rfd/XXXX-upgrade-system.md

```diff
@@ -0,0 +1,1222 @@
+---
+authors: Forrest Marshall (forrest@goteleport.com)
+state: draft
+---
+
+# RFD XXXX - Upgrade System
+
+## What
+
+System for automatic upgrades of teleport installations.
+
+
+## Why
+
+Teleport must be periodically updated in order to integrate security patches.  Regular
+updates also ensure that users can take advantage of improvements in stability and
+performance.  Outdated teleport installations impose additional burdens on us and on
+our users. Teleport does not currently assist with upgrades in any way, and the burden
+of manual upgrades can be prohibitive.
+
+Reducing the friction of upgrades is beneficial both in terms of security and user
+experience. Doing this may also indirectly lower our own support load.
+
+
+## Intro
+
+### Suggested Reading
+
+While not required, it is helpful to have some familiarity with [The Update Framework](https://theupdateframework.com/)
+when reading this RFD. TUF is a flexible framework for securing upgrade systems. It provides a robust framework
+for key rotation, censorship detection, package validation, and much more.
+
+
+### High-Level Goals
```

Comment:
Who is our target customer that we're building this for? The ones who run Teleport directly on VMs w/ systemd? Those who run in k8s with helm? Those who run our AMIs or docker images? None of these? All of them?

I think calling this out early will help readers understand the motivation behind the decisions we're making here. Plus, it's often best to solve a smaller problem really well than to do an okay job at solving a larger problem.

Reply:
May also require some mention of different installation angles, such as if the customer uses RPM. If this is just a binary on disk update, there may be some interactions with packaging systems, like the version of the package not matching the binary, or if it will just call rpm/deb/etc to update.